### PR TITLE
Feat/gba 39 Unit test for common package

### PR DESCRIPTION
**Pull request by jesusdmedinac**

### CODE REVIEW

File: app/src/main/java/com/mupper/gobus/commons/Generics.kt [hunk 1]
```diff
@@ -11,22 +12,22 @@ import kotlin.reflect.KClass
  * Created by jesus.medina on 01/2020.
  * Mupper
  */
-fun <VDB : ViewDataBinding> newInstance(
+fun <VDB : ViewDataBinding> newStepInstance(
```
Comment (proyectox123) [inline app/src/main/java/com/mupper/gobus/commons/Generics.kt:15]: May I suggest the used of _apply_?

`
fun <VDB : ViewDataBinding> newStepInstance(
        stepFragment: StepFragment<VDB>,
        @LayoutRes layoutResId: Int,
        args: Bundle
) = stepFragment.apply {
    stepFragment.arguments = args.apply {
        putInt(LAYOUT_RESOURCE_ID_ARG_KEY, layoutResId)
    }
}
`

File: app/src/test/java/com/mupper/gobus/commons/extension/ContextKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.Test
+import org.mockito.BDDMockito.willDoNothing
+import org.mockito.BDDMockito.willReturn
+
+@Suppress("DEPRECATION")
+class ContextKtTest {
+    @Test
+    fun `app should return applicationContext as an instance of GobusApp`() {
+        // GIVEN
+        val mockContext: Context = mock()
```
Comment (proyectox123) [inline app/src/test/java/com/mupper/gobus/commons/extension/ContextKtTest.kt]: Would it be good moving some mocks like mockContext and mockResources to setUp(Before) function?

File: app/src/test/java/com/mupper/gobus/commons/scope/ScopeImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.containsString
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Test
+
+class ScopeImplTest {
+    @Test
+    fun `ScopeImpl should set uiDispatcher as given dispatcher`() {
+        // GIVEN
+        val expectedDispatcher = Dispatchers.Default
```
Comment (proyectox123) [inline app/src/test/java/com/mupper/gobus/commons/scope/ScopeImplTest.kt]: Would it be good moving expectedDispatcher to setUp(Before) function?

File: app/src/test/java/com/mupper/gobus/commons/scope/ScopeImplTest.kt [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        scopeImpl.initScope()
+
+        // THEN
+        assertThat(scopeImpl.coroutineContext.toString(), containsString("DefaultDispatcher"))
+    }
+
+    @Test(expected = UninitializedPropertyAccessException::class)
+    fun `coroutineContext should throw UninitializedPropertyAccessException if job has not been initialized`() {
+        // GIVEN
+        val scopeImpl = Scope.Impl(Dispatchers.Unconfined)
```
Comment (proyectox123) [inline app/src/test/java/com/mupper/gobus/commons/scope/ScopeImplTest.kt:57]: Would it be good moving scopeImpl to setUp(Before) function?

File: app/src/main/java/com/mupper/gobus/commons/Generics.kt (hunk 1 above)
Comment (jesusdmedinac) [reply app/src/main/java/com/mupper/gobus/commons/Generics.kt:15]: That's a good one! Thanks!

File: app/src/test/java/com/mupper/gobus/commons/extension/ContextKtTest.kt (hunk 2 above)
Comment (jesusdmedinac) [reply app/src/test/java/com/mupper/gobus/commons/extension/ContextKtTest.kt]: I think I forget to do the refactor 😅

File: app/src/main/java/com/mupper/gobus/commons/Generics.kt (hunk 1 above)
Comment (jesusdmedinac) [reply app/src/main/java/com/mupper/gobus/commons/Generics.kt:15]: Next time could you use of code suggestion to allow me to apply directly in your comment:
image

**Pull request merged**